src/main/java/seedu/application/model/job/Status.java

```diff
@@ -0,0 +1,81 @@
+package seedu.application.model.job;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.application.commons.util.AppUtil;
+
+/**
+ * Represents a Job's status in the application book.
+ */
+public class Status {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Status is not case sensitive and should only be in the form: TO_BE_SUBMITTED, PENDING, APPROVED, REJECTED";
+
+    public static final String IN_PROGRESS = JobStatus.TO_BE_SUBMITTED.toString();
+    public static final Status DEFAULT_STATUS = new Status(IN_PROGRESS);
+
+    public final String status;
+
+    /**
+     * The enumeration representing the status of a job application.
+     * Possible values are:
+     *   TO_BE_SUBMITTED: The job application is yet to be submitted.
+     *   PENDING: The job application is pending review or processing.
+     *   APPROVED: The job application has been approved.
+     *   REJECTED: The job application has been rejected.
+     */
+    public enum JobStatus {
+        TO_BE_SUBMITTED, PENDING, APPROVED, REJECTED
```

Comment:
You can consider adding other status such as SHORTLISTED, INTERVIEWED, UNDER_REVIEW, DECLINED, WITHDRAWN